src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```
@@ -635,64 +635,76 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
+                Commands::QueryImageResponse::Type delegateResponse;
+                ConvertFromQueryImageResponseParms(data, delegateResponse);
+
+                // If update is not available, return the delegate response
+                if (!hasUpdate) {
+                    handler->AddResponse(cachedCommandPath, delegateResponse);
+                    handle.Release();
+                    return;
+                }
+
+                // If there is an update available, return a response with status - Available and set the image Uri in the
+                // response
+                auto fabricIndex = handler->GetSubjectDescriptor().fabricIndex;
+                auto nodeId = handler->GetSubjectDescriptor().subject;
+                CHIP_ERROR err = gOtaSender.PrepareForTransfer(fabricIndex, nodeId);
+                if (CHIP_NO_ERROR != err) {
+
+                    // Handle busy error separately as we have a query image response status that maps to busy
+                    if (err == CHIP_ERROR_BUSY) {
+                        ChipLogError(Controller, "Resource is busy handling another request.");
+                        Commands::QueryImageResponse::Type response;
+                        response.status = static_cast<OTAQueryStatus>(MTROTASoftwareUpdateProviderOTAQueryStatusBusy);
+                        response.delayedActionTime.SetValue(delegateResponse.delayedActionTime.ValueOr(kDelayedActionTimeSeconds));
+                        handler->AddResponse(cachedCommandPath, response);
                         handle.Release();
-                        gOtaSender.ResetState();
                         return;
                     }
-
-                    response.imageURI.SetValue(uri);
-                    handler->AddResponse(cachedCommandPath, response);
+                    LogErrorOnFailure(err);
+                    handler->AddStatus(cachedCommandPath, StatusIB(err).mStatus);
```

This is fine, but now I am very confused by this comment in `ConfigureState`:
```
        // The caller of this method maps CHIP_ERROR to specific BDX Status Codes (see GetBdxStatusCodeFromChipError)
        // and those are used by the BDX session to prepare the status report.
```
This code is the only way `ConfigureState` is reached, right?  And we're not calling GetBdxStatusCodeFromChipError here....

---

I think we do that for aborttransfer not prepare transfer anymore. I need to modify the comment accordingly

---

This should just return the err now. fixed it.